src/coreclr/System.Private.CoreLib/src/System/Runtime/DependentHandle.cs

```diff
@@ -0,0 +1,224 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime
+{
+    /// <summary>
+    /// Represents a dependent GC handle, which will conditionally keep a dependent object instance alive
+    /// as long as a target object instance is alive as well, without representing a strong reference to the
+    /// target object instance. That is, a <see cref="DependentHandle"/> value with a given object instance as
+    /// target will not cause the target to be kept alive if there are no other strong references to it, but
+    /// it will do so for the dependent object instance as long as the target is alive.
+    /// <para>
+    /// This type is conceptually equivalent to having a weak reference to a given target object instance A, with
+    /// that object having a field or property (or some other strong reference) to a dependent object instance B.
+    /// </para>
+    /// </summary>
+    public struct DependentHandle : IDisposable
```

Comment:
Should we add a [DebuggerDisplay(...)] attribute?  And/or a [DebuggerTypeProxy(...)]?

Reply:
Not sure how to structure them exactly, as in, what info did you have in mind for them to expose that wouldn't already be displayed by the built-in debugger view? I guess I can see how eg. `[DebuggerDisplay]` could add more info, but wouldn't a `[DebuggerTypeProxy]` with the same `IsAllocated`, `Target` and `Dependent` properties offer the same info as what VS would already do on its own? Or is that to avoid throwing `InvalidOperationExceptions` in the debugger if the handle is not allocated, or something else? Thanks! 🙂

Reply:
I was actually thinking more about, for example,  the Target and Dependent properties showing as null instead of throwing an exception if the instance isn't allocated.  But, not a big deal.